### ISSUE TITLE
Unable to determine owner and permissions on remote sftp mount

### ISSUE THREAD

**Issue by urandom0** [elementary/files#1879]
### What Happened?

Elementary Files can't determine the owner and permissions on mounted remote sftp share. In folder Properties window -> Permissions tab it says:
Owner: ---
Group: ---

And I cant paste file on share. It works with Nautilis and other file manager.

### Steps to Reproduce

1. Mount sftp share
2. Copy and paste any file

### Expected Behavior

It paste file.

### OS Version

6.x (Odin)

### Software Version

Latest release (I have run all updates)

**Comment by jeremypw** [comment]
On my system, connecting to another Linux machine on the local network the properties window shows user as "--" as you describe but gives the group as "1000" and gives sensible numbers for read/write/execute permissions e.g. "755" or "644".  I can copy/paste files across the connection without difficulty.  So it may be to do with the configuration of the server.  Also Files requires the remote system to have a "native" filesystem in order to get the permissions - Is the remote system running Linux? 

If Nautilus is unaffected then there does appear to be an issue.

**Comment by urandom0** [comment]
@jeremypw 
"Is the remote system running Linux?"

Yes, but it is not my own. I can't change the configuration on it.
The Nautilus work as expected.

**Comment by jeremypw** [comment]
A workaround for this is to enter right click on the folder icon where the files of interest reside on the remote server and chooser "Open in ... Terminal".  You can then type `ls -alh` to list the files and their permissions and you can use `chmod` to alter them (if you have the right).

**Comment by jeremypw** [comment]
I'll try and find to look into it.  Tbh most work has been done on local filesystems.

**Comment by jeremypw** [comment]
Update - using `ls -alh` on the gvfs mapped folder as suggested does not, in fact, work as expected.  It always gives the owner and group as the local user - not the user and group on the remote server.   So far I have managed to fix Files to the extent that it gets the correct UID and GID for the remote files. However, it cannot get the remote user name or group name from these IDs.  However, using Nautilus v 3.36.3, does not yield the names either - it substitutes the ids.

Below are images of the permissions of a file on a remote SSH server which was created by a user other than the one which has logged in. In neither case is the ability to alter the user and group is offered.

## Files (after fix) ##
![FilesSSHPermissions](https://user-images.githubusercontent.com/10513844/139715049-b8af8666-cbe3-47ee-8055-8b445437f4fc.png)

## Nautilus 3.36.3 ##
![NautilusSSHPermissions](https://user-images.githubusercontent.com/10513844/139715070-c3114a1d-a376-47b0-a9bd-8cae598851c4.png)

The file could only be read under these conditions.  When logged in as the same user then the file could be edited.

**Comment by urandom0** [comment]
Left: Files
Right: Nautilus

Same sftp, same folder.
![1](https://user-images.githubusercontent.com/93508883/139908817-754a01bf-5180-4920-9e0f-7dd576c64fc9.jpg)

I can't paste file with Files:
![2](https://user-images.githubusercontent.com/93508883/139908878-8396b6ec-1fdd-4488-bb07-b15ee7b16230.jpg)

I CAN paste file with Nautilus:
![3](https://user-images.githubusercontent.com/93508883/139908924-ee1d15b2-f4c8-4e35-ab4a-5f9b76cf2a8e.jpg)

**Comment by urandom0** [comment]
It is another sftp server. Same effect:
Left: Files
Right: Nautilus

![2_1](https://user-images.githubusercontent.com/93508883/139909575-60a60bfd-3241-43b5-a8fa-4bd5ac09fa68.jpg)

I can't paste with Files:
![2_2](https://user-images.githubusercontent.com/93508883/139910173-eca0c515-98c1-4bc7-9346-c3a0342bf468.jpg)

I can paste with Nautilus:
![2_3](https://user-images.githubusercontent.com/93508883/139910211-d22f920e-595f-4652-a755-1b805cf5de33.jpg)

**Comment by jeremypw** [comment]
@urandom0 Are you able to build and try out the linked PR?

**Comment by urandom0** [comment]
> @urandom0 Are you able to build and try out the linked PR?

How I can do that?

**Comment by jeremypw** [comment]
OK, if your not familiar with building from source it doesnt matter.  However, if you wish to try then there is *basic* information in the README on https://github.com/elementary/files.  There is further information on the elementary developer website: https://docs.elementary.io/develop/

**Comment by urandom0** [comment]
@jeremypw 
I built it but I get error when I try to run:
io.elementary.files: error while loading shared libraries: libpantheon-files-widgets.so.6: cannot open shared object file: No such file or directory

**Comment by urandom0** [comment]
Okay, I solved that by installing libpantheon-files-widgets0. But the permission error still exists.

**Comment by jeremypw** [comment]
Thanks for trying!

Hmm, the PR pass CI testing and it runs OK on my system.

Use `git branch` to check you are on the `fix-ssh-info` branch.

On building there will be some "warning" messages but all the dependencies should be found and you should end up with a "Compilation succeeded - 6 warnings" message.

Did you install with `sudo ninja install` before running with `io.elementary.files`?  Or did you run locally with `./src/io.elementary.files`?  

Does `sudo ninja test` pass?

You will need to kill any existing Files processes before running the new version as the PR does change a library file. If uncertain then log out and back in.

If you install the PR branch then you should re-install either the master branch or the stable release when you are finished testing it.

**Comment by jeremypw** [comment]
I usually build (from the build directory) with `sudo ninja` - not sure if that makes a difference though.

**Comment by urandom0** [comment]
Here is the complete build process (dependencies installed previously):

---------------------------------------

```
urandom0@StandardPCQ35ICH92009097659c7:~$ git clone https://github.com/elementary/files.git
Cloning into 'files'...
remote: Enumerating objects: 73283, done.
remote: Counting objects: 100% (7760/7760), done.
remote: Compressing objects: 100% (2037/2037), done.
remote: Total 73283 (delta 7217), reused 6140 (delta 5712), pack-reused 65523
Receiving objects: 100% (73283/73283), 153.26 MiB | 6.13 MiB/s, done.
Resolving deltas: 100% (60579/60579), done.
urandom0@StandardPCQ35ICH92009097659c7:~$ cd files
urandom0@StandardPCQ35ICH92009097659c7:~/files$ meson build --prefix=/usr
The Meson build system
Version: 0.53.2
Source dir: /home/urandom0/files
Build dir: /home/urandom0/files/build
Build type: native build
Project name: io.elementary.files
Project version: 6.0.4
C compiler for the host machine: cc (gcc 9.3.0 "cc (Ubuntu 9.3.0-17ubuntu1~20.04) 9.3.0")
C linker for the host machine: cc ld.bfd 2.34
Vala compiler for the host machine: valac (valac 0.48.19)
Host machine cpu family: x86_64
Host machine cpu: x86_64
Found pkg-config: /usr/bin/pkg-config (0.29.1)
Run-time dependency glib-2.0 found: YES 2.64.6
Run-time dependency gobject-2.0 found: YES 2.64.6
Run-time dependency gio-2.0 found: YES 2.64.6
Run-time dependency gio-unix-2.0 found: YES 2.64.6
Run-time dependency gmodule-2.0 found: YES 2.64.6
Run-time dependency gee-0.8 found: YES 0.20.3
Run-time dependency gtk+-3.0 found: YES 3.24.20
Run-time dependency granite found: YES 6.1.2
Run-time dependency libhandy-1 found: YES 1.2.0
Run-time dependency pango found: YES 1.44.7
Run-time dependency sqlite3 found: YES 3.31.1
Run-time dependency libcanberra found: YES 0.30
Run-time dependency libnotify found: YES 0.7.9
Library posix found: YES
Library linux found: YES
Library m found: YES
Run-time dependency plank found: YES 0.11.89
Run-time dependency zeitgeist-2.0 found: YES 1.0.2
Configuring config.h using configuration
Found pkg-config: /usr/bin/pkg-config (0.29.1)
Library pantheon-files-core-C found: YES
Configuring io.elementary.files.service using configuration
Configuring io.elementary.files.Filemanager1.service using configuration
Configuring io.elementary.files-pkexec using configuration
Configuring io.elementary.files.policy.in using configuration
Configuring io.elementary.files.desktop.in using configuration
Configuring io.elementary.files.appdata.xml.in using configuration
WARNING: Gettext not found, all translation targets will be ignored.
Run-time dependency cloudproviders found: YES 0.3.0
Run-time dependency libgit2-glib-1.0 found: YES 0.28.0.1
Run-time dependency gdk-x11-3.0 found: YES 3.24.20
Run-time dependency gdk-wayland-3.0 found: YES 3.24.20
Dependency gdk-x11-3.0 found: YES 3.24.20 (cached)
Dependency gdk-wayland-3.0 found: YES 3.24.20 (cached)
Run-time dependency systemd found: YES 245
Configuring io.elementary.files.xdg-desktop-portal.service using configuration
Configuring org.freedesktop.impl.portal.desktop.elementary.files.service using configuration
Program meson/post_install.py found: YES (/usr/bin/python3 /home/urandom0/files/meson/post_install.py)
Build targets in project: 18

Found ninja-1.10.0 at /usr/bin/ninja
urandom0@StandardPCQ35ICH92009097659c7:~/files$ cd build
urandom0@StandardPCQ35ICH92009097659c7:~/files/build$ ninja
[90/149] Compiling Vala source ../filechooser-portal/ExternalWindow.vala ../filechooser-portal/FileCho.../filechooser-portal/Main.vala libwidgets/pantheon-files-widgets.vapi libcore/pantheon-files-core.vapi.
../filechooser-portal/ExternalWindow.vala:80.41-80.44: warning: Argument 1: Cannot pass null to non-null parameter type
        x11_display = Gdk.Display.open (null);
                                        ^^^^
../filechooser-portal/ExternalWindow.vala:81.35-81.38: warning: Argument 1: Cannot pass null to non-null parameter type
        Gdk.set_allowed_backends (null);
                                  ^^^^
../filechooser-portal/ExternalWindow.vala:115.45-115.48: warning: Argument 1: Cannot pass null to non-null parameter type
        wayland_display = Gdk.Display.open (null);
                                            ^^^^
../filechooser-portal/ExternalWindow.vala:116.35-116.38: warning: Argument 1: Cannot pass null to non-null parameter type
        Gdk.set_allowed_backends (null);
                                  ^^^^
../filechooser-portal/LegacyFileChooserDialog.vala:65.39-65.68: warning: `Gtk.Dialog.get_action_area' has been deprecated since 3.12
../filechooser-portal/Main.vala:360.13-360.23: warning: local variable `session_bus' declared but never used
        var session_bus = GLib.Bus.get_sync (GLib.BusType.SESSION);
            ^^^^^^^^^^^
Compilation succeeded - 6 warning(s)
[149/149] Linking target filechooser-portal/io.elementary.files.xdg-desktop-portal.
urandom0@StandardPCQ35ICH92009097659c7:~/files/build$ sudo ninja install
[0/1] Installing files.
Installing libcore/libpantheon-files-core.so.6.0.4 to /usr/lib64
Installing libcore/pantheon-files-core.h to /usr/include/pantheon-files-core
Installing libcore/pantheon-files-core.vapi to /usr/share/vala/vapi
Installing libwidgets/libpantheon-files-widgets.so.6.0.4 to /usr/lib64
Installing libwidgets/pantheon-files-widgets.h to /usr/include
Installing libwidgets/pantheon-files-widgets.vapi to /usr/share/vala/vapi
Installing src/io.elementary.files to /usr/bin
Installing pantheon-files-daemon/io.elementary.files-daemon to /usr/bin
Installing plugins/contractor/libpantheon-files-contractor.so to /usr/lib64/io.elementary.files/plugins/core
Installing plugins/pantheon-files-trash/libpantheon-files-trash.so to /usr/lib64/io.elementary.files/plugins/core
Installing plugins/pantheon-files-ctags/libpantheon-files-ctags.so to /usr/lib64/io.elementary.files/plugins/core
Installing plugins/pantheon-files-cloud/libpantheon-files-cloudproviders.so to /usr/lib64/io.elementary.files/plugins/core
Installing plugins/git/libpantheon-files-git.so to /usr/lib64/io.elementary.files/plugins/core
Installing plugins/send-by-email/libpantheon-files-send-by-email.so to /usr/lib64/io.elementary.files/plugins/core
Installing filechooser-portal/io.elementary.files.xdg-desktop-portal to /usr/libexec
Installing /home/urandom0/files/libcore/marlin-file-operations.h to /usr/include/pantheon-files-core
Installing /home/urandom0/files/build/meson-private/pantheon-files-core.pc to /usr/lib64/pkgconfig
Installing /home/urandom0/files/build/meson-private/pantheon-files-widgets.pc to /usr/lib64/pkgconfig
Installing /home/urandom0/files/build/data/io.elementary.files.service to /usr/share/dbus-1/services
Installing /home/urandom0/files/build/data/io.elementary.files.Filemanager1.service to /usr/share/dbus-1/services
Installing /home/urandom0/files/build/data/io.elementary.files-pkexec to /usr/bin
Installing /home/urandom0/files/data/icons/poof.png to /usr/share/pixmaps/io.elementary.files
Installing /home/urandom0/files/data/schemas/io.elementary.files.gschema.xml to /usr/share/glib-2.0/schemas
Installing /home/urandom0/files/plugins/contractor/pantheon-files-contractor.plug to /usr/lib64/io.elementary.files/plugins/core
Installing /home/urandom0/files/plugins/pantheon-files-trash/pantheon-files-trash.plug to /usr/lib64/io.elementary.files/plugins/core
Installing /home/urandom0/files/plugins/pantheon-files-ctags/pantheon-files-ctags.plug to /usr/lib64/io.elementary.files/plugins/core
Installing /home/urandom0/files/plugins/pantheon-files-cloud/pantheon-files-cloudproviders.plug to /usr/lib64/io.elementary.files/plugins/core
Installing /home/urandom0/files/plugins/git/pantheon-files-git.plug to /usr/lib64/io.elementary.files/plugins/core
Installing /home/urandom0/files/plugins/send-by-email/pantheon-files-send-by-email.plug to /usr/lib64/io.elementary.files/plugins/core
Installing /home/urandom0/files/build/filechooser-portal/io.elementary.files.xdg-desktop-portal.service to /usr/lib/systemd/user
Installing /home/urandom0/files/filechooser-portal/io.elementary.files.portal to /usr/share/xdg-desktop-portal/portals
Installing /home/urandom0/files/build/filechooser-portal/org.freedesktop.impl.portal.desktop.elementary.files.service to /usr/share/dbus-1/services
Running custom install script '/usr/bin/python3 /home/urandom0/files/meson/post_install.py'
Compiling gsettings schemas...
Updating icon cache...
Updating desktop database...
urandom0@StandardPCQ35ICH92009097659c7:~/files/build$ 
```

---------------------------------------

_"Did you install with sudo ninja install before running with io.elementary.files? Or did you run locally with ./src/io.elementary.files?"_

I installed it with "sudo ninja install".

_"You will need to kill any existing Files processes before running the new version as the PR does change a library file. If uncertain then log out and back in."_
    
I uninstalled elementary-files packages before build (but I didn't logout - I will try this).

_"Does sudo ninja test pass?"_
    
urandom0@StandardPCQ35ICH92009097659c7:~/files/build$ sudo ninja test
[0/1] Running all tests.
1/4 UtilTests                               OK       0.03 s 
2/4 MarlinIconInfoTests                     OK       0.79 s 
3/4 GOFFileTests                            OK       0.06 s 
4/4 GOFDirectoryAsyncTests                  OK       0.48 s 

Ok:                    4
Expected Fail:         0
Fail:                  0
Unexpected Pass:       0
Skipped:               0
Timeout:               0

Full log written to /home/urandom0/files/build/meson-logs/testlog.txt
```

**Comment by jeremypw** [comment]
@urandom0 Looks like you built and installed the master version successfully?  I do not see any errors reported in the above log.  To test the PR you will need to enter `git checkout fix-ssh-info` and repeat the `sudo ninja install` command. Then type `io.elementary.files` to run the installed version.   If it exits immediately it means you still have a Files window open - close all Files windows first.

**Comment by urandom0** [comment]
_"Looks like you built and installed the master version successfully?"_

Yes.

```
urandom0@StandardPCQ35ICH92009097659c7:~/files/build$ git checkout fix-ssh-info
Branch 'fix-ssh-info' set up to track remote branch 'fix-ssh-info' from 'origin'.
Switched to a new branch 'fix-ssh-info'
urandom0@StandardPCQ35ICH92009097659c7:~/files/build$ sudo ninja install
[sudo] password for urandom0:     
[0/1] Regenerating build files.
The Meson build system
Version: 0.53.2
Source dir: /home/urandom0/files
Build dir: /home/urandom0/files/build
Build type: native build
Project name: io.elementary.files
Project version: 6.0.4
C compiler for the host machine: cc (gcc 9.3.0 "cc (Ubuntu 9.3.0-17ubuntu1~20.04) 9.3.0")
C linker for the host machine: cc ld.bfd 2.34
Vala compiler for the host machine: valac (valac 0.48.19)
Host machine cpu family: x86_64
Host machine cpu: x86_64
Dependency glib-2.0 found: YES 2.64.6 (cached)
Dependency gobject-2.0 found: YES 2.64.6 (cached)
Dependency gio-2.0 found: YES 2.64.6 (cached)
Dependency gio-unix-2.0 found: YES 2.64.6 (cached)
Dependency gmodule-2.0 found: YES 2.64.6 (cached)
Dependency gee-0.8 found: YES 0.20.3 (cached)
Dependency gtk+-3.0 found: YES 3.24.20 (cached)
Dependency granite found: YES 6.1.2 (cached)
Dependency libhandy-1 found: YES 1.2.0 (cached)
Dependency pango found: YES 1.44.7 (cached)
Dependency sqlite3 found: YES 3.31.1 (cached)
Dependency libcanberra found: YES 0.30 (cached)
Dependency libnotify found: YES 0.7.9 (cached)
Library posix found: YES
Library linux found: YES
Library m found: YES
Dependency plank found: YES 0.11.89 (cached)
Dependency zeitgeist-2.0 found: YES 1.0.2 (cached)
Configuring config.h using configuration
Found pkg-config: /usr/bin/pkg-config (0.29.1)
Library pantheon-files-core-C found: YES
Configuring io.elementary.files.service using configuration
Configuring io.elementary.files.Filemanager1.service using configuration
Configuring io.elementary.files-pkexec using configuration
Configuring io.elementary.files.policy.in using configuration
Configuring io.elementary.files.desktop.in using configuration
Configuring io.elementary.files.appdata.xml.in using configuration
WARNING: Gettext not found, all translation targets will be ignored.
Dependency cloudproviders found: YES 0.3.0 (cached)
Dependency libgit2-glib-1.0 found: YES 0.28.0.1 (cached)
Dependency gdk-x11-3.0 found: YES 3.24.20 (cached)
Dependency gdk-wayland-3.0 found: YES 3.24.20 (cached)
Dependency gdk-x11-3.0 found: YES 3.24.20 (cached)
Dependency gdk-wayland-3.0 found: YES 3.24.20 (cached)
Dependency systemd found: YES 245 (cached)
Configuring io.elementary.files.xdg-desktop-portal.service using configuration
Configuring org.freedesktop.impl.portal.desktop.elementary.files.service using configuration
Program meson/post_install.py found: YES (/usr/bin/python3 /home/urandom0/files/meson/post_install.py)
Build targets in project: 18

Found ninja-1.10.0 at /usr/bin/ninja
[30/87] Compiling Vala source ../filechooser-portal/ExternalWindow.vala ../filechooser-portal/FileChoo.../filechooser-portal/Main.vala libwidgets/pantheon-files-widgets.vapi libcore/pantheon-files-core.vapi.
../filechooser-portal/ExternalWindow.vala:80.41-80.44: warning: Argument 1: Cannot pass null to non-null parameter type
        x11_display = Gdk.Display.open (null);
                                        ^^^^
../filechooser-portal/ExternalWindow.vala:81.35-81.38: warning: Argument 1: Cannot pass null to non-null parameter type
        Gdk.set_allowed_backends (null);
                                  ^^^^
../filechooser-portal/ExternalWindow.vala:115.45-115.48: warning: Argument 1: Cannot pass null to non-null parameter type
        wayland_display = Gdk.Display.open (null);
                                            ^^^^
../filechooser-portal/ExternalWindow.vala:116.35-116.38: warning: Argument 1: Cannot pass null to non-null parameter type
        Gdk.set_allowed_backends (null);
                                  ^^^^
../filechooser-portal/LegacyFileChooserDialog.vala:65.39-65.68: warning: `Gtk.Dialog.get_action_area' has been deprecated since 3.12
../filechooser-portal/Main.vala:360.13-360.23: warning: local variable `session_bus' declared but never used
        var session_bus = GLib.Bus.get_sync (GLib.BusType.SESSION);
            ^^^^^^^^^^^
Compilation succeeded - 6 warning(s)
[34/35] Installing files.
Installing libcore/libpantheon-files-core.so.6.0.4 to /usr/lib64
Installing libcore/pantheon-files-core.h to /usr/include/pantheon-files-core
Installing libcore/pantheon-files-core.vapi to /usr/share/vala/vapi
Installing libwidgets/libpantheon-files-widgets.so.6.0.4 to /usr/lib64
Installing libwidgets/pantheon-files-widgets.h to /usr/include
Installing libwidgets/pantheon-files-widgets.vapi to /usr/share/vala/vapi
Installing src/io.elementary.files to /usr/bin
Installing pantheon-files-daemon/io.elementary.files-daemon to /usr/bin
Installing plugins/contractor/libpantheon-files-contractor.so to /usr/lib64/io.elementary.files/plugins/core
Installing plugins/pantheon-files-trash/libpantheon-files-trash.so to /usr/lib64/io.elementary.files/plugins/core
Installing plugins/pantheon-files-ctags/libpantheon-files-ctags.so to /usr/lib64/io.elementary.files/plugins/core
Installing plugins/pantheon-files-cloud/libpantheon-files-cloudproviders.so to /usr/lib64/io.elementary.files/plugins/core
Installing plugins/git/libpantheon-files-git.so to /usr/lib64/io.elementary.files/plugins/core
Installing plugins/send-by-email/libpantheon-files-send-by-email.so to /usr/lib64/io.elementary.files/plugins/core
Installing filechooser-portal/io.elementary.files.xdg-desktop-portal to /usr/libexec
Installing /home/urandom0/files/libcore/marlin-file-operations.h to /usr/include/pantheon-files-core
Installing /home/urandom0/files/build/meson-private/pantheon-files-core.pc to /usr/lib64/pkgconfig
Installing /home/urandom0/files/build/meson-private/pantheon-files-widgets.pc to /usr/lib64/pkgconfig
Installing /home/urandom0/files/build/data/io.elementary.files.service to /usr/share/dbus-1/services
Installing /home/urandom0/files/build/data/io.elementary.files.Filemanager1.service to /usr/share/dbus-1/services
Installing /home/urandom0/files/build/data/io.elementary.files-pkexec to /usr/bin
Installing /home/urandom0/files/data/icons/poof.png to /usr/share/pixmaps/io.elementary.files
Installing /home/urandom0/files/data/schemas/io.elementary.files.gschema.xml to /usr/share/glib-2.0/schemas
Installing /home/urandom0/files/plugins/contractor/pantheon-files-contractor.plug to /usr/lib64/io.elementary.files/plugins/core
Installing /home/urandom0/files/plugins/pantheon-files-trash/pantheon-files-trash.plug to /usr/lib64/io.elementary.files/plugins/core
Installing /home/urandom0/files/plugins/pantheon-files-ctags/pantheon-files-ctags.plug to /usr/lib64/io.elementary.files/plugins/core
Installing /home/urandom0/files/plugins/pantheon-files-cloud/pantheon-files-cloudproviders.plug to /usr/lib64/io.elementary.files/plugins/core
Installing /home/urandom0/files/plugins/git/pantheon-files-git.plug to /usr/lib64/io.elementary.files/plugins/core
Installing /home/urandom0/files/plugins/send-by-email/pantheon-files-send-by-email.plug to /usr/lib64/io.elementary.files/plugins/core
Installing /home/urandom0/files/build/filechooser-portal/io.elementary.files.xdg-desktop-portal.service to /usr/lib/systemd/user
Installing /home/urandom0/files/filechooser-portal/io.elementary.files.portal to /usr/share/xdg-desktop-portal/portals
Installing /home/urandom0/files/build/filechooser-portal/org.freedesktop.impl.portal.desktop.elementary.files.service to /usr/share/dbus-1/services
Running custom install script '/usr/bin/python3 /home/urandom0/files/meson/post_install.py'
Compiling gsettings schemas...
Updating icon cache...
Updating desktop database...
urandom0@StandardPCQ35ICH92009097659c7:~/files/build$ 

```
Now I see the owner and group of folder:

![3_1](https://user-images.githubusercontent.com/93508883/140176795-236e72d1-5fa2-4131-8015-1cde04e934c0.jpg)

But I still can't paste file (it works with Nautilus) and I can't save file with Code (works with Gedit):
![3_2](https://user-images.githubusercontent.com/93508883/140178097-b999878a-9ea9-41a2-8528-a00dd45e0f16.jpg)

**Comment by jeremypw** [comment]
OK, thanks.  Would it be possible for you to split this into two separate issues so that at least one can be marked as fixed (once the PR is merged)?  Thanks for your effort to test!

**Comment by jeremypw** [comment]
I notice that "Everyone" does not have write access according to the properties so this may be why changes cannot be saved to the original file.  Not sure why it is working with Gedit though - maybe it is using the "owner" permissions.  When you connected did you connect as the user who owns the file or as the local user?

**Comment by urandom0** [comment]
I connect with remote user who owns the file (my id is 1038 on the server).
Sometimes I connect with Far Manager from Windows, in this I can save file. Sometimes I connect with Winscp, in this case it works too. Sometimes I connect with Nautilus from Manjaro - it is works too, and it is works with Nautilus and Gedit on Elementary OS.
But it doesn't works with Files or Code.
